editor/animation_track_editor.cpp

```diff
@@ -5300,11 +5331,95 @@ void AnimationTrackEditor::_new_track_node_selected(NodePath p_path) {
+	Changes the Animation's path to match that node.
+*/
+void AnimationTrackEditor::_move_track_to_node(NodePath p_path, int p_track) {
+	// Get new_node
+	Node *tree_root = EditorNode::get_singleton()->get_tree()->get_root();
+	Node *new_node = tree_root->get_node_or_null(p_path);
+	ERR_FAIL_NULL(new_node);
+
+	// Get AnimationPlayer's root_node
+	AnimationPlayer *ap = AnimationPlayerEditor::get_singleton()->get_player();
+	if (!ap) {
+		ERR_FAIL_EDMSG("No AnimationPlayer is currently being edited.");
+	}
+	Node *root_node = ap->get_node_or_null(ap->get_root_node());
+	if (!root_node) {
+		EditorNode::get_singleton()->show_warning(TTR("Not possible to change node path without a root"));
+		return;
+	}
+
+	// Get New Path, in respect to AP's root_node
+	String current_track_path = animation->track_get_path(p_track);
+	String property_path = current_track_path.get_slice(":", 1);
+	String new_path = root_node->get_path_to(new_node);
+	new_path = new_path + ":" + property_path;
+
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	undo_redo->create_action(TTR("Change Track Path"));
+	undo_redo->add_do_method(animation.ptr(), "track_set_path", p_track, new_path);
+	undo_redo->add_undo_method(animation.ptr(), "track_set_path", p_track, current_track_path);
+	undo_redo->commit_action();
+}
+
+void AnimationTrackEditor::_change_track_property_selected(const String &p_name, int p_track) {
+	String current_track_path = animation->track_get_path(p_track);
+	String base_path = current_track_path.get_slice(":", 0);
+	String new_path = base_path + ":" + p_name;
+
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	undo_redo->create_action(TTR("Change Track Path"));
+	undo_redo->add_do_method(animation.ptr(), "track_set_path", p_track, new_path);
+	undo_redo->add_undo_method(animation.ptr(), "track_set_path", p_track, current_track_path);
+	undo_redo->commit_action();
+}
+
+void AnimationTrackEditor::_change_track_target_property_pressed(int p_track) {
+	AnimationPlayer *ap = AnimationPlayerEditor::get_singleton()->get_player();
+	if (!ap) {
+		ERR_FAIL_EDMSG("No AnimationPlayer is currently being edited.");
+	}
+	NodePath current_path = animation->track_get_path(p_track);
+	Node *root_node = ap->get_node_or_null(ap->get_root_node());
+	Node *current_node = root_node->get_node_or_null(animation->track_get_path(p_track));
+	if (!current_node) {
+		EditorNode::get_singleton()->show_warning(TTR("Not possible to change property without a valid target node"));
+		return;
+	}
+	// Reconnect here to bind track index
+	if (change_prop_selector->is_connected("selected", callable_mp(this, &AnimationTrackEditor::_change_track_property_selected))) {
+		change_prop_selector->disconnect("selected", callable_mp(this, &AnimationTrackEditor::_change_track_property_selected));
+	}
+	change_prop_selector->connect("selected", callable_mp(this, &AnimationTrackEditor::_change_track_property_selected).bind(p_track));
```

Comment:
It would be nice to get rid of this connecting/disconnecting logic. You can carry the information about selected track by using a new variable (or reusing one from the existing dialogs).